File: Data Structures & Algorithms/single-number/submission-0.java
class Solution {
    public int singleNumber(int[] nums) {
        //11
        //10
        //11
        int ans = nums[0];
        //= 3
        /// 10
        //7 = 0111
        //6 = 0110
        //^ = 0001
        //6 = 0110
        //^ = 0111
        //7 = 0111
        //^ = 0000
        //8 = 1000
        //^ = 1000
        for(int i = 1; i < nums.length; i++){
            ans = nums[i] ^ ans;
        }
        return ans;
    }
}

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        curSet = []
        def backtrack(i):
            nonlocal res
            if i >= len(nums):
                res.append(curSet[:])
                return
            curSet.append(nums[i])
            backtrack(i+1)
            curSet.pop()
            backtrack(i+1)

        backtrack(0)
        return res
            
